### DISCUSSION TITLE
question about clamscan / clamdscan

### DISCUSSION THREAD

**Discussion by squalou** [ivangabriele/clamav-desktop#169] [Questions & Answers]
Hi,

this is more a question than an issue, I didn't know where to submit it.
ClamTK being unmaintained I'm looking for alternatives for the future and yours is promising.

Unfortunately I didn't manage to check myself, so here's the question.

Is it be possible to use `clamdscan` instead of `clamscan` for manual scans using UI ? By a setting somewhere.

(I have to use `clamdscan --fdpass --multiscan` to have decent perfs, and never found the option on clamtk, I manually patched the perl script underneath :) )

thanks !

**Comment by ivangabriele** [comment]
Hi @squalou and sorry for the late reply,

That's actually touching one of my potential pain points, which I won't solve yet in the next release.

In ClamAV Desktop I orchestrate `clamscan` to scan files but my caveat is that it loads the virus definitions each time it's run. `clamd` daemon seems to have an underlying strategy to somehow preload these definitions which improves the scanning boot time. Even though for large scans, listing file is definitely what takes ALOT of time (another pain I plan on optimizing in futures releases, if possible).

Now, in the context of running from the CLI without any GUI, and with an existing `clamd` daemon service running, `clamdscan` is preferable over `clamscan` to run you scans since it will leverage its optimizations ^^.
